utils_linux.go

```diff
@@ -196,11 +197,20 @@ func (r *runner) run(config *specs.Process) (int, error) {
 		r.destroy()
 		return -1, err
 	}
+
+	if len(r.listenFDs) > 0 && r.preserveFDs > 0 {
+		return -1, fmt.Errorf("Preserve FDs and socket activation (LISTEN_FDS) are mutually incompatible")
+	}
```

Comment:
@crosbymichael thanks, so to check I understand, you are suggesting that `--preserve-fds` should be cumulative over `SD_LISTEN` (or anything else which extends `process.ExtraFiles` in the future). IOW given `LISTEN_FDS=3` and `--preserve-fds=5` then we would preserve `[3, 11)` (in addition to stdio). That's 3, 4 & 5 from `LISTEN_FDS=3` and 6, 7, 8, 9 & 10 from `--preserve-fds=5`.

I'll do that.